src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -5506,9 +5506,6 @@ private VariableDeclaratorSyntax ParseVariableDeclarator(
             // specifically treats it as a variable name, even if it could be interpreted as a
             // keyword.
             var name = this.ParseIdentifierToken();
-            BracketedArgumentListSyntax argumentList = null;
-            EqualsValueClauseSyntax initializer = null;
```

Comment:
keeping track of when these were set, and hte invariants around them was veyr difficult to reason about.